chall01/osalmine.py

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/python3
+
+import sys
+
+morse = { 'A':'.-', 'B':'-...',
+                    'C':'-.-.', 'D':'-..', 'E':'.',
+                    'F':'..-.', 'G':'--.', 'H':'....',
+                    'I':'..', 'J':'.---', 'K':'-.-',
+                    'L':'.-..', 'M':'--', 'N':'-.',
+                    'O':'---', 'P':'.--.', 'Q':'--.-',
+                    'R':'.-.', 'S':'...', 'T':'-',
+                    'U':'..-', 'V':'...-', 'W':'.--',
+                    'X':'-..-', 'Y':'-.--', 'Z':'--..'}
+
+if len(sys.argv) != 2:
+	print("usage: ./osalmine.py <a-zA-Z string>")
+	quit()
+
+string = sys.argv[1].upper();
+if len(string) == 0:
+	print("usage: ./osalmine.py <a-zA-Z string>")
+	quit()
+
+if not all(x.isalpha() or x.isspace() for x in string):
+	print("usage: ./osalmine.py <a-zA-Z string>")
+	quit()
+
+for letter in string:
+	if (letter == " "):
+		print(" ", end = '')
+	else:
+		print(morse[letter], end = '');
```

Comment:
There is a so-called pythonic way to write code in Python. Which in practice means using all the tools provided to write laconic and readable code. You can read more about pythonic yourself, I'll only provide an example in this particular case:
`print(*(morse[letter] for letter in string), sep="")`
works if you added space to keys of morse. if not, would look a bit more ugly:
`print(*(morse[letter] if letter != " " else " " for letter in string), sep="")`